csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs

```diff
@@ -75,13 +73,21 @@ BuildScript GetNugetRestoreScript() =>
                         QuoteArgument(projectOrSolution.FullPath).
                         Argument("-DisableParallelProcessing").
                         Script;
+
+                BuildScript GetMonoVersionScript() => new CommandBuilder(builder.Actions).
```

Comment:
Maybe rename to `IsMonoInstalled` (even though an exit code is returned - the important thing is whether `mono` is installed)?

Would it be worth extending the `IBuildActions` interface (with a method returning af bool) instead of making a script that is immediately executed?

Then it is possible to make an implementation like
```
public bool UseDotnet() => IsRunningOnAppleSilicon() || (!IsWIndows() && !IsMonoInstalled())
```
which can be used in `MsBuildCommand` and on l. 90.

Reply:
Thank you. I pushed a commit with your idea.